### PR TITLE
Add json_array_length Presto Json function

### PR DESCRIPTION
**Pull request by pramodsatya**
Adds Json function json_array_length to find the length of Json arrays, if the Json object doesn't represent an array, return NULL.

### CODE REVIEW

File: velox/functions/prestosql/JsonFunctions.h [hunk 1]
```diff
@@ -46,4 +46,23 @@ struct JsonExtractScalarFunction {
   }
 };
 
+// json_array_length(json) -> bigint
+// Returns the array length of json (a string containing a JSON array)
+// Returns 0 if json is not an array.
```
Comment (kagamiori) [inline velox/functions/prestosql/JsonFunctions.h:51]: `json_array_length` should return `NULL` when the input json does not represent an array.

Also, let's add a description of this function in velox/docs/functions/json.rst and remove the comment here. Feel free to copy the description from Presto's documentation: https://prestodb.io/docs/current/functions/json.html?highlight=json_array_length#json_array_length.

**Review by kagamiori — COMMENTED** [review]
Hi Pramod, Thank you for adding this function.

File: velox/functions/prestosql/JsonFunctions.h [hunk 2]
```diff
@@ -46,4 +46,23 @@ struct JsonExtractScalarFunction {
...
+template <typename T>
+struct jsonArrayLengthFunction {
```
Comment (aditi-pandit) [inline velox/functions/prestosql/JsonFunctions.h:50]: Nit : Uppercase 'J' in the name.

File: velox/functions/prestosql/tests/JsonFunctionsTest.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      json_array_length(
+          R"([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20])"),
+      20);
+
+  EXPECT_EQ(json_array_length(R"(1)"), 0);
+  EXPECT_EQ(json_array_length(R"("hello")"), 0);
+  EXPECT_EQ(json_array_length(R"("")"), 0);
+  EXPECT_EQ(json_array_length(R"(true)"), 0);
+  EXPECT_EQ(json_array_length(R"({"k1":"v1"})"), 0);
+  EXPECT_EQ(json_array_length(R"({"k1":[0,1,2]})"), 0);
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/JsonFunctionsTest.cpp]: If we have a json that is a struct which one field an array and one not, then that should return NULL also. Please can you add such a test case.

File: velox/functions/prestosql/JsonFunctions.h (hunk 1 above)
Comment (pramodsatya) [reply velox/functions/prestosql/JsonFunctions.h]: Thanks, modified accordingly

**Review by aditi-pandit — COMMENTED** [review]
Thanks Pramod. Looks good overall.

File: velox/functions/prestosql/tests/JsonFunctionsTest.cpp (hunk 3 above)
Comment (pramodsatya) [reply velox/functions/prestosql/tests/JsonFunctionsTest.cpp]: Added, thanks!

File: velox/functions/prestosql/JsonFunctions.h [hunk 4]
```diff
@@ -46,4 +46,20 @@ struct JsonExtractScalarFunction {
...
+template <typename T>
+struct JsonArrayLengthFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      int64_t& result,
+      const arg_type<Varchar>& json) {
+    auto parsedJson = folly::parseJson(json);
+    if(!parsedJson.isArray()){
+      result = int64_t(NULL);
```
Comment (kagamiori) [inline velox/functions/prestosql/JsonFunctions.h:58]: Hi Pramod, `void call(...)` indicates this function never returns null. To make it return null, write it as `bool call(...)` instead. Then return true if the current result is not null, or return false if the current result is null (and no need to assign anything to `result` if it is supposed to be null). (An explanation can be found right above the second code example here: https://facebookincubator.github.io/velox/develop/scalar-functions.html#simple-functions.)

Comment (pramodsatya) [reply velox/functions/prestosql/JsonFunctions.h]: Thanks for sharing, modified accordingly

**Comment by kagamiori** [comment]
Looks good to me. Thank you for making the changes!

**Review by kagamiori — APPROVED** [review]

**Comment by bikramSingh91** [comment]
@pramodsatya Thanks for making this change. As a follow up can you please also update the coverage map.
Instructions: https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/coverage/README.md

cc: @kagamiori

**Pull request closed without merging**